### PR TITLE
fix: workerthreads support in testkit

### PR DESCRIPTION
**Pull request by olehrakwix**

### CODE REVIEW

File: packages/electron-commons/test/initializers/electron-node.spec.ts [hunk 1]
```diff
@@ -35,9 +35,9 @@ describe('onDisconnectHandler for node environment initializer', () => {
             });
 
             await dispose();
-
-            await expect(exitPromise).to.eventually.deep.eq({ exitCode: 0 });
+            await expect(exitPromise).to.eventually.be.fulfilled;
```
Comment (AviVahl) [inline packages/electron-commons/test/initializers/electron-node.spec.ts:38]: previous is better. why switch?

File: packages/runtime-node/src/worker-thread-entry.ts [hunk 2]
```diff
@@ -53,7 +53,11 @@ const handleStartupMessage = async (command: WorkerThreadStartupCommand) => {
         childEnvName: environmentContextName,
     });
 
-    disposeNodeEnv = runningNodeEnv.dispose;
+    disposeNodeEnv = () => {
+        ensureWorkerThreadContext(parentPort);
```
Comment (AviVahl) [inline packages/runtime-node/src/worker-thread-entry.ts:57]: this was already validated during start. use `!` in next line instead?

File: packages/runtime-node/src/worker-thread-entry.ts [hunk 3]
```diff
@@ -65,7 +69,7 @@ const messageHandler = (message: unknown) => {
             break;
 
         case 'workerThreadDisposeCommand':
-            Promise.all([disposeNodeEnv])
+            Promise.all([disposeNodeEnv()])
```
Comment (AviVahl) [inline packages/runtime-node/src/worker-thread-entry.ts:72]: is the `Promise.all` even needed?

File: packages/test-kit/test/run-environment.spec.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        const { dispose, runningApi } = await getRunningFeature({
+            env: serverEnv,
+            feature: workerThreadFeature,
+            featureName: workerThreadFeature.id,
+            basePath: featurePath,
+        });
+
+        const result = await runningApi.workerService.initAndCallWorkerEcho('hello');
+        expect(result).to.equal('hello from worker');
+        await dispose();
```
Comment (AviVahl) [inline packages/test-kit/test/run-environment.spec.ts:21]: if expect above fails, this won't be called. use createDisposables

File: packages/test-kit/package.json [hunk 5]
```diff
@@ -4,7 +4,9 @@
   "main": "dist/index.js",
   "types": "dist/index.d.ts",
   "scripts": {
-    "test": "mocha \"test/**/*.unit.ts?(x)\""
+    "test": "yarn test:unit && yarn test:spec",
```
Comment (AviVahl) [inline packages/test-kit/package.json:7]: why is the separation needed? call the test file `.unit`

**Review by AviVahl — CHANGES_REQUESTED** [review]

File: packages/electron-commons/test/initializers/electron-node.spec.ts (hunk 1 above)
Comment (olehrakwix) [reply packages/electron-commons/test/initializers/electron-node.spec.ts:38]: it gets flaky for some reasons. somehow on CI it sometimes has { signal: SIGTERM }, and others { exitCode: 0 }
does not reproduce locally

File: packages/runtime-node/src/worker-thread-entry.ts (hunk 2 above)
Comment (olehrakwix) [reply packages/runtime-node/src/worker-thread-entry.ts]: fixed

File: packages/test-kit/test/run-environment.spec.ts (hunk 4 above)
Comment (olehrakwix) [reply packages/test-kit/test/run-environment.spec.ts]: done

File: packages/test-kit/package.json (hunk 5 above)
Comment (olehrakwix) [reply packages/test-kit/package.json]: ok, renamed

**Review by AviVahl — APPROVED** [review]

**Pull request merged**